src/instance_launcher.ts

```diff
@@ -63,23 +88,37 @@ export default class InstanceLauncher {
         const currentInventory = await this.jibriTracker.getCurrent(ctx, groupName);
         const count = currentInventory.length;
 
-        if (count < group.scalingOptions.desiredCount && count < group.scalingOptions.maxDesired) {
-            ctx.logger.info('[Launcher] Will scale up to the desired count', { groupName, desiredCount, count });
+        // set stat for current count of instances
+        instancesCount.set({ group: group.name }, count);
+        try {
+            if (count < group.scalingOptions.desiredCount && count < group.scalingOptions.maxDesired) {
+                ctx.logger.info('[Launcher] Will scale up to the desired count', { groupName, desiredCount, count });
+
+                const actualScaleUpQuantity =
+                    Math.min(group.scalingOptions.maxDesired, group.scalingOptions.desiredCount) - count;
+                await this.cloudManager.scaleUp(ctx, group, count, actualScaleUpQuantity);
 
-            const actualScaleUpQuantity =
-                Math.min(group.scalingOptions.maxDesired, group.scalingOptions.desiredCount) - count;
-            await this.cloudManager.scaleUp(ctx, group, count, actualScaleUpQuantity);
-        } else if (count > group.scalingOptions.desiredCount && count > group.scalingOptions.minDesired) {
-            ctx.logger.info('[Launcher] Will scale down to the desired count', { groupName, desiredCount, count });
+                // increment launched instance stats for the group
+                instancesLaunched.inc({ group: group.name }, actualScaleUpQuantity);
+            } else if (count > group.scalingOptions.desiredCount && count > group.scalingOptions.minDesired) {
+                ctx.logger.info('[Launcher] Will scale down to the desired count', { groupName, desiredCount, count });
 
-            const actualScaleDownQuantity =
-                count - Math.max(group.scalingOptions.minDesired, group.scalingOptions.desiredCount);
+                const actualScaleDownQuantity =
```

Comment:
Are these unintended whitespace changes?

Reply:
I moved this whole block into a try {} to catch the errors by group, so this was intentional.